### PR TITLE
Bluetooth: audio: shell: Add command to set location

### PR DESCRIPTION
**Pull request by szymon-czapracki**
Allow setting audio locations from shell.

### CODE REVIEW

**Comment by szymon-czapracki** [comment]
@MariuszSkamra FYI

File: subsys/bluetooth/shell/audio.c [hunk 1]
```diff
@@ -1409,6 +1409,33 @@ static int cmd_term_broadcast_sink(const struct shell *sh, size_t argc,
...
+	if (!strcmp(argv[1], "sink")) {
+		dir = BT_AUDIO_DIR_SINK;
+	} else if (!strcmp(argv[1], "source")) {
+		dir = BT_AUDIO_DIR_SOURCE;
+	} else {
+		shell_error(sh, "Unsupported dir: %s", argv[1]);
+		return -ENOEXEC;
+	}
+
+	loc = shell_strtoul(argv[2], 10, &err);
```
Comment (Thalley) [inline subsys/bluetooth/shell/audio.c:1429]: Given that is a bitmask, should we consider using base 16 instead of base 10?

**Review by alwa-nordic — DISMISSED** [review]

**Comment by Thalley** [comment]
@szymon-czapracki Please resolve the failing CI checks

File: subsys/bluetooth/shell/audio.c [hunk 2]
```diff
@@ -1568,6 +1595,9 @@ SHELL_STATIC_SUBCMD_SET_CREATE(audio_cmds,
 #endif /* CONFIG_BT_AUDIO_UNICAST */
 	SHELL_CMD_ARG(send, NULL, "Send to Audio Stream [data]",
 		      cmd_send, 1, 1),
+	SHELL_CMD_ARG(set_location, NULL,
```
Comment (MariuszSkamra) [inline subsys/bluetooth/shell/audio.c:1598]:
```suggestion
	SHELL_COND_CMD_ARG(CONFIG_BT_AUDIO_CAPABILITY, set_location, NULL,
```

**Review by MariuszSkamra — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/shell/audio.c (hunk 1 above)
Comment (szymon-czapracki) [reply subsys/bluetooth/shell/audio.c]: Done

File: subsys/bluetooth/shell/audio.c (hunk 2 above)
Comment (szymon-czapracki) [reply subsys/bluetooth/shell/audio.c]: Done

File: subsys/bluetooth/shell/audio.c [hunk 3]
```diff
@@ -1579,6 +1606,9 @@ SHELL_STATIC_SUBCMD_SET_CREATE(audio_cmds,
 #endif /* CONFIG_BT_AUDIO_UNICAST */
 	SHELL_CMD_ARG(send, NULL, "Send to Audio Stream [data]",
 		      cmd_send, 1, 1),
+	SHELL_COND_CMD_ARG(CONFIG_BT_AUDIO_CAPABILITY, set_location, NULL,
+			"<direction: sink, source> <location bitmask>",
+			cmd_set_loc, 3, 0),
```
Comment (Thalley) [inline subsys/bluetooth/shell/audio.c:1611]: Indentation
```suggestion
	SHELL_COND_CMD_ARG(CONFIG_BT_AUDIO_CAPABILITY, set_location, NULL,
			   "<direction: sink, source> <location bitmask>",
			   cmd_set_loc, 3, 0),
```

File: subsys/bluetooth/shell/audio.c [hunk 4]
```diff
@@ -1409,6 +1409,33 @@ static int cmd_term_broadcast_sink(const struct shell *sh, size_t argc,
...
+	if (!strcmp(argv[1], "sink")) {
+		dir = BT_AUDIO_DIR_SINK;
+	} else if (!strcmp(argv[1], "source")) {
+		dir = BT_AUDIO_DIR_SOURCE;
+	} else {
+		shell_error(sh, "Unsupported dir: %s", argv[1]);
+		return -ENOEXEC;
+	}
+
+	loc = shell_strtoul(argv[2], 16, &err);
```
Comment (Thalley) [inline subsys/bluetooth/shell/audio.c:1429]: Remember to initialize `err` to `0` first :)

File: subsys/bluetooth/shell/audio.c [hunk 5]
```diff
@@ -1409,6 +1409,33 @@ static int cmd_term_broadcast_sink(const struct shell *sh, size_t argc,
 }
 #endif /* CONFIG_BT_AUDIO_BROADCAST_SINK */
 
+static int cmd_set_loc(const struct shell *sh, size_t argc, char *argv[])
+{
+	int err;
+	enum bt_audio_dir dir;
+	enum bt_audio_location loc;
+
+	ctx_shell = sh;
```
Comment (Thalley) [inline subsys/bluetooth/shell/audio.c:1418]:
```suggestion
```

**Review by Thalley — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/shell/audio.c (hunk 4 above)
Comment (szymon-czapracki) [reply subsys/bluetooth/shell/audio.c:1427]: Done

**Review by Thalley — APPROVED** [review]

**Review by MariuszSkamra — APPROVED** [review]

**Pull request merged**